libs/platform/src/lib/components/form/select/select/select.component.ts

```diff
@@ -0,0 +1,321 @@
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    Host,
+    Inject,
+    OnInit,
+    Optional,
+    Self,
+    SkipSelf,
+    TemplateRef,
+    ViewChild,
+    ViewEncapsulation,
+    Input,
+    EventEmitter,
+    Output,
+    AfterViewChecked
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+
+import { DynamicComponentService, FdSelectChange, MobileModeConfig, SelectControlState, PopoverFillMode } from '@fundamental-ngx/core';
+import { SelectDataSource, DATA_PROVIDERS, DataProvider } from '../../../../domain/data-source';
+import { OptionItem } from '../../../../domain';
+import { FormField } from '../../form-field';
+import { FormFieldControl } from '../../form-control';
+import { BaseSelect, FdpSelectionChangeEvent } from '../commons/base-select';
+import { SelectConfig } from '../select.config';
+
+@Component({
+    selector: 'fdp-select',
+    templateUrl: './select.component.html',
+    styleUrls: ['./select.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [{ provide: FormFieldControl, useExisting: SelectComponent, multi: true }]
+})
+export class SelectComponent extends BaseSelect implements OnInit, AfterViewInit, AfterViewChecked {
```

Comment:
I am not sure if select shoudl have DataSource. DataSource are for components that requires to handle larger datasets. and for Select we had List binding